profiling/src/lib.rs

```diff
@@ -339,6 +312,67 @@ pub struct RequestLocals {
     pub vm_interrupt_addr: *const AtomicBool,
 }
 
+/// take a sample every X bytes
+#[cfg(feature = "allocation_profiling")]
+const ALLOCATION_PROFILING_INTERVAL: f32 = 1024.0 * 50.0;
```

Comment:
NIT: Maybe add a comment how this value was picked? This causes 10x more frequent sampling compared to Go. Unless PHP programs have a 10x small allocation rate, this would cause 10x more overhead.

It's also fine to just add a comment that this value is temporary and will be replaced with a better one later.

Last but not least: Have you considered making this value user configurable? That's what Go does, see [runtime.MemProfileRate](https://pkg.go.dev/runtime#MemProfileRate).

Reply:
Added a comment and created a new JIRA ticket for making the value configurable, thanks for the input 🙇